dfastmi/batch/AnalyserDflowfm.py

```diff
@@ -0,0 +1,376 @@
+from pathlib import Path
+from dfastmi.batch.SedimentationVolume import comp_sedimentation_volume
+from dfastmi.kernel.core import main_computation, dzq_from_du_and_h
+from dfastmi.batch.DflowfmLoggers import AnalyserDflowfmLogger
+from dfastmi.batch.OutputDataDflowfm import OutputDataDflowfm
+from dfastmi.batch.XykmData import XykmData
+from dfastmi.io.GridOperations import GridOperations
+from dfastmi.kernel.typehints import Vector, BoolVector
+from shapely.geometry.linestring import LineString
+
+import numpy
+import os
+from typing import Any, Dict, Optional, TextIO, Tuple, Union
+
+class AnalyserDflowfm():
+    """
+    Class that analyses the Dflowfm data.
+    """
+    
+    _logger : AnalyserDflowfmLogger
+    
+    def __init__(self, display : bool, report : TextIO, needs_tide : bool, old_zmin_zmax : bool, outputdir : Path):
+        """
+        Arguments
+        ---------
+        display : bool
+            Flag indicating text output to stdout.
+        report : TextIO
+            Text stream for log file.
+        needs_tide : bool
+            Specifies whether the tidal boundary is needed.
+        old_zmin_zmax : bool
+            Specifies the minimum and maximum should follow old or new definition.
+        outputdir : Path
+            Path of output directory.
+        """
+        self._logger = AnalyserDflowfmLogger(display, report)
+        self._needs_tide = needs_tide
+        self._old_zmin_zmax = old_zmin_zmax
+        self._outputdir = outputdir
+    
+    def analyse(self, 
+                q_threshold : float,
```

Comment:
Can ConfigInitializer object be used here?

Reply:
Can you maybe set some argument to class fields? like filenames for instance. Is easier to understand the argument list of private calls

Reply:
up to what point should I make a difference between class fields and passed in parameters?